examples/single-sign-on/iam-identity-center/sso.tf

```diff
@@ -0,0 +1,262 @@
+  name             = "EKSClusterAdmin"
+  description      = "Amazon EKS Cluster Admins."
+  instance_arn     = tolist(data.aws_ssoadmin_instances.this.arns)[0]
+  relay_state      = "https://s3.console.aws.amazon.com/s3/home?region=us-west-2#"
+  session_duration = "PT1H"
+}
+
+resource "aws_ssoadmin_permission_set" "user" {
+  name             = "EKSClusterUser"
+  description      = "Amazon EKS Cluster Users."
+  instance_arn     = tolist(data.aws_ssoadmin_instances.this.arns)[0]
+  relay_state      = "https://s3.console.aws.amazon.com/s3/home?region=us-west-2#"
+  session_duration = "PT1H"
+}
+
+data "aws_iam_policy_document" "admin" {
+  statement {
+    sid = "EKSAdmin"
+    actions = [
+      "eks:*"
+    ]
+    resources = [
+      module.eks.cluster_arn
+    ]
+  }
+  statement {
+    sid = "AssumeRole"
+    actions = [
+      "sts:AssumeRole"
+    ]
+    resources = [
+      "*"
+    ]
+  }
+}
+
+data "aws_iam_policy_document" "user" {
+  statement {
+    sid = "EKSRead"
+    actions = [
+      "eks:List*",
+      "eks:Describe*",
+      "eks:AccessKubernetesApi"
+    ]
+    resources = [
+      module.eks.cluster_arn
+    ]
+  }
+
+  statement {
+    sid = "AssumeRole"
+    actions = [
+      "sts:AssumeRole"
+    ]
+    resources = [
+      "*"
+    ]
+  }
+}
+
+resource "aws_ssoadmin_permission_set_inline_policy" "admin" {
+  inline_policy      = data.aws_iam_policy_document.admin.json
+  instance_arn       = tolist(data.aws_ssoadmin_instances.this.arns)[0]
+  permission_set_arn = aws_ssoadmin_permission_set.admin.arn
+}
+
+resource "aws_ssoadmin_permission_set_inline_policy" "user" {
+  inline_policy      = data.aws_iam_policy_document.user.json
+  instance_arn       = tolist(data.aws_ssoadmin_instances.this.arns)[0]
+  permission_set_arn = aws_ssoadmin_permission_set.user.arn
+}
+
+resource "aws_ssoadmin_managed_policy_attachment" "admin" {
+  instance_arn       = tolist(data.aws_ssoadmin_instances.this.arns)[0]
+  managed_policy_arn = "arn:aws:iam::aws:policy/PowerUserAccess"
+  permission_set_arn = aws_ssoadmin_permission_set.admin.arn
+}
+
+resource "aws_ssoadmin_managed_policy_attachment" "user" {
+  instance_arn       = tolist(data.aws_ssoadmin_instances.this.arns)[0]
+  managed_policy_arn = "arn:aws:iam::aws:policy/job-function/ViewOnlyAccess"
+  permission_set_arn = aws_ssoadmin_permission_set.user.arn
+}
+
+resource "aws_identitystore_user" "admin" {
+  identity_store_id = tolist(data.aws_ssoadmin_instances.this.identity_store_ids)[0]
+  display_name      = "Platform Admin"
+  user_name         = "admin@example.com"
+
+  name {
+    family_name = "Admin"
+    given_name  = "Platform"
+  }
+}
+
+resource "aws_identitystore_user" "user" {
+  identity_store_id = tolist(data.aws_ssoadmin_instances.this.identity_store_ids)[0]
+  display_name      = "Developer User"
+  user_name         = "user@example.com"
+
+  name {
+    family_name = "User"
+    given_name  = "Developer"
+  }
+}
```

Comment:
Would it make sense to accept a map of User configuration as a variable?

Reply:
I think yes, my concern is to the example become "consumable".

Reply:
Done! :)